engineering/frontend/code-style.md

```diff
@@ -0,0 +1,84 @@
+# Code style
+
+While the vast majority of base code style is enforced by [ESLint](https://eslint.org/) and [Prettier](https://prettier.io/), please also where possible, stick to the contribution guidelines below. These rules should be kept in mind when reviewing Pull Requests:
+
+- Code should be functional in style rather than Object Orientated or Imperative unless there are no clean alternatives.
+
+  - Use pure functions where possible to make them testable and modular.
+  - Avoid mutating variables and the `let` keyword.
+  - React Components should be stateless functional components where possible.
+  - Avoid classes and stateful modules where possible.
+  - Don't Repeat Yourself. Make extensive use of the constants and utils files for re-usable strings and methods.
+  - The above rules can be relaxed for test scripts.
+
+- React components should be simple and composable and cater to real life UI design problems:
+
+  - **Simplicity**: Strive to keep the component API fairly simple and show real-world scenarios of using the component.
+  - **Representational**: React components should be templates, free of logic, and purely presentational. It aims to make our components shareable and easy to test.
+
+  - **Composition**: Break down components into smaller parts with minimal props to keep complexity low, and compose them together. This will ensure that the styles and functionality are flexible and extensible.
+
+  - **Accessibility**: When creating a component, keep accessibility top of mind. This includes keyboard navigation, focus management, color contrast, voice over, and the correct aria-\* attributes.
+
+  - **Naming Props**: We all know naming is the hardest thing in this industry. Generally, ensure a prop name is indicative of what it does. Boolean props should be named using auxiliary verbs such as does, has, is and should. For example, Button uses `isDisabled`, `isLoading`, etc.
```

Comment:
To me I prefer we remove not using the `is`
verb mean boolean props/state
`isDisabled -> disabled` the HTML standard using disabled, readonly for boolean
`isLoading -> loading`